org-roam-graph.el

```diff
@@ -144,120 +141,109 @@ into a digraph."
-           (format "  \"%s\" [%s];\n"
-                   file
-                   (->> node-properties
-                        (mapcar (lambda (n)
-                                  (concat (car n) "=" "\"" (cdr n) "\"")))
-                        (s-join ","))))))
+           (format "  \"%s\" [%s];\n" file
+                   (mapconcat (lambda (n)
+                                (org-roam-graph--dot-option n nil "\""))
+                              node-properties ",")))))
       (dolist (edge edges)
-        (insert (format "  \"%s\" -> \"%s\";\n"
-                        (xml-escape-string (car edge))
-                        (xml-escape-string (cadr edge)))))
-
+        (insert (apply #'format `("  \"%s\" -> \"%s\";\n"
+                                  ,@(mapcar #'xml-escape-string edge)))))
       (insert (format "  edge [%s];\n"
-                      (->> org-roam-graph-edge-cites-extra-config
-                           (mapcar (lambda (n)
-                                     (concat (car n) "=" (cdr n))))
-                           (s-join ","))))
+                      (mapconcat #'org-roam-graph--dot-option
+                                 org-roam-graph-edge-cites-extra-config ",")))
       (dolist (edge edges-cites)
-        (insert (format "  \"%s\" -> \"%s\";\n"
-                        (xml-escape-string (car edge))
-                        (xml-escape-string (cadr edge)))))
+        (insert (apply #'format `("  \"%s\" -> \"%s\";\n"
+                                  ,@(mapcar #'xml-escape-string edge)))))
       (insert "}")
       (buffer-string))))
 
-(defun org-roam-graph-build (&optional node-query)
-  "Generate a graph showing the relations between nodes in NODE-QUERY.
-For building and showing the graph in a single step see `org-roam-graph-show'."
-  (interactive)
+(defun org-roam-graph--build (&optional node-query)
+  "Generate a graph showing the relations between nodes in NODE-QUERY."
   (unless org-roam-graph-executable
     (user-error "Can't find %s executable.  Please check if it is in your path"
                 org-roam-graph-executable))
   (let* ((node-query (or node-query
                          `[:select [file titles]
                                    :from titles
                                    ,@(org-roam-graph--expand-matcher 'file t)]))
-         (graph      (org-roam-graph--build node-query))
+         (graph      (org-roam-graph--dot node-query))
          (temp-dot   (make-temp-file "graph." nil ".dot" graph))
          (temp-graph (make-temp-file "graph." nil ".svg")))
     (call-process org-roam-graph-executable nil 0 nil
                   temp-dot "-Tsvg" "-o" temp-graph)
     temp-graph))
 
+(defun org-roam-graph--build-connected-component (file &optional max-distance)
+  "Build a graph of nodes connected to FILE.
+If MAX-DISTANCE is non-nil, limit nodes to MAX-DISTANCE steps."
+  (unless (org-roam--org-roam-file-p file)
+    (user-error "%s is not an org-roam file" file))
```

Comment:
Maybe put quotes around the file name in the message? Also, in the case of a non-file-visiting buffer the message will read `nil is not an...`, which might be confusing?

Reply:
Good point. I thought on this and think it would actually make sense to move the error checking to the `org-roam-graph` command, since that is the user entry-point. I'll add a case for each condition there so we can give more helpful user-error messages.